projects/packages/my-jetpack/_inc/components/product-card/use-pricing-data.ts

```diff
@@ -41,34 +46,65 @@ const getPurchaseAction = ( detail: ProductCamelCase ) => {
 
 	if ( detail.status === PRODUCT_STATUSES.CAN_UPGRADE || isUpgradable ) {
 		if ( upgradeHasPrice ) {
-			return __( 'Upgrade', 'jetpack-my-jetpack' );
+			return { label: __( 'Upgrade', 'jetpack-my-jetpack' ), onClick: onCheckout };
 		}
 		return null;
 	}
 
-	return __( 'Purchase', 'jetpack-my-jetpack' );
+	return { label: __( 'Purchase', 'jetpack-my-jetpack' ), onClick: onCheckout };
 };
 
-const getLearnMoreAction = ( detail: ProductCamelCase ) => {
+const getSecondaryAction = ( detail: ProductCamelCase, onActivate: () => void ) => {
 	const isNotActiveOrNeedsExplicitFreePlan =
 		! detail.isPluginActive || detail.status === PRODUCT_STATUSES.NEEDS_PURCHASE_OR_FREE;
 
 	if (
 		isNotActiveOrNeedsExplicitFreePlan &&
 		( detail.tiers.includes( 'free' ) || detail.hasFreeOffering )
 	) {
-		return __( 'Start for free', 'jetpack-my-jetpack' );
+		return {
+			label: __( 'Start for free', 'jetpack-my-jetpack' ),
+			onClick: onActivate,
+		};
 	}
 
-	return __( 'Learn more', 'jetpack-my-jetpack' );
+	return { label: __( 'Learn more', 'jetpack-my-jetpack' ), href: `#/add-${ detail.slug }` };
 };
 
 const usePricingData = ( slug: string ) => {
 	const { detail } = useProduct( slug );
+	const { wpcomProductSlug, ...data } = parsePricingData( detail.pricingForUi );
+
+	const { isUserConnected } = useMyJetpackConnection();
+	const { myJetpackUrl, siteSuffix } = getMyJetpackWindowInitialState();
+	const { activate, isPending: isActivating } = useActivate( slug );
+	const { run: runCheckout, hasCheckoutStarted } = useProductCheckoutWorkflow( {
+		from: 'my-jetpack',
+		productSlug: wpcomProductSlug,
+		redirectUrl: myJetpackUrl,
+		connectAfterCheckout: ! isUserConnected,
+		siteSuffix,
+	} );
+
+	const handleActivate = useCallback( () => {
+		activate( {} );
```

Comment:
I think we can just have `activate()`, I see that used that way elsewhere 🤔

Reply:
I think I messed up the typing of useSimpleMutation last time I was working on this, I'll fix it in the cleanup PR, or log it as maintenance task if it's more complex.